src/node_buffer.cc

```diff
@@ -835,7 +835,9 @@ void Compare(const FunctionCallbackInfo<Value> &args) {
 
   size_t cmp_length = MIN(obj_a_length, obj_b_length);
 
-  int32_t val = memcmp(obj_a_data, obj_b_data, cmp_length);
+  int32_t val = 0;
```

Comment:
Technically, yes (it's indeed the _exact same code_). But this is not clear code style.

--

```javascript
int32_t val;
if (cmp_length > 0) {
    val = memcmp(obj_a_data, obj_b_data, cmp_length);
} else {
    val = 0;
}
``` 
or
```javascript
int32_t val;
if (cmp_length > 0)
    val = memcmp(obj_a_data, obj_b_data, cmp_length);
else
    val = 0;
```
or
```javascript
int32_t val = cmp_length > 0 ? memcmp(obj_a_data, obj_b_data, cmp_length) : 0;
```
all are readed as «set `val` to `memcmp(…)` if `cmp_length` is not zero, else set it to `0`.

--

```javascript
int32_t val = 0;
if (cmp_length > 0)
    val = memcmp(obj_a_data, obj_b_data, cmp_length);
```
on the first read looks like «set `val` to zero _by default_, and if `cmp_length > 0`, set it to `memcmp(…)`».

The first issue is that line `int32_t val = 0;` in that context means «treat the compared `Buffer` objects as being equal by default», and the `if` below means «and then do some checks».

The second issue is that an accidental edit of that place could miss the fact that this is a single logical block, and could insert something in between of `int32_t val = 0;` and the `if (…`. It might be still technically correct (and from the point of view that you described it would still do «make sure that val is initialized, do something, set val if …»), but that would make the code harder to read (you will have to jump between several separated lines to gasp one simple logical construction). This also could lead to some error in further edits.